src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ExactMethodInstantiationsNode.cs

```diff
@@ -0,0 +1,180 @@
+using Internal.TypeSystem;
+using Internal.NativeFormat;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Hashtable of all exact (non-canonical) generic method instantiations compiled in the module.
+    /// </summary>
+    internal sealed class ExactMethodInstantiationsNode : ObjectNode, ISymbolNode
+    {
+        private ObjectAndOffsetSymbolNode _endSymbol;
+        private ExternalReferencesTableNode _externalReferences;
+
+        private NativeWriter _nativeWriter;
+        private Section _nativeSection;
+        private VertexHashtable _hashtable;
+
+        private HashSet<MethodDesc> _visitedMethods;
+        private List<MethodDesc> _exactMethodInstantiationsList;
+
+        public ExactMethodInstantiationsNode(ExternalReferencesTableNode externalReferences)
+        {
+            _endSymbol = new ObjectAndOffsetSymbolNode(this, 0, "__exact_method_instantiations_End", true);
+            _externalReferences = externalReferences;
+
+            _nativeWriter = new NativeWriter();
+            _hashtable = new VertexHashtable();
+            _nativeSection = _nativeWriter.NewSection();
+            _nativeSection.Place(_hashtable);
+
+            _visitedMethods = new HashSet<MethodDesc>();
+            _exactMethodInstantiationsList = new List<MethodDesc>();
+        }
+
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(nameMangler.CompilationUnitPrefix).Append("__exact_method_instantiations");
+        }
+
+        public ISymbolNode EndSymbol => _endSymbol;
+        public int Offset => 0;
+        public override bool IsShareable => false;
+        public override ObjectNodeSection Section => ObjectNodeSection.DataSection;
+        public override bool StaticDependenciesAreComputed => true;
+        protected override string GetName() => this.GetMangledName();
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+            // Dependencies for this node are tracked by the method code nodes
+            if (relocsOnly)
+                return new ObjectData(Array.Empty<byte>(), Array.Empty<Relocation>(), 1, new ISymbolNode[] { this });
+
+            // Zero out the hashset so that we AV if someone tries to insert after we're done.
+            _visitedMethods = null;
+
+            // Ensure the native layout data has been saved, in order to get valid Vertex offsets for the signature Vertices
+            factory.MetadataManager.NativeLayoutInfo.SaveNativeLayoutInfoWriter(factory);
+
+            foreach (MethodDesc method in _exactMethodInstantiationsList)
+            {
+                bool getUnboxingStub = method.OwningType.IsValueType && !method.Signature.IsStatic;
+                IMethodNode methodEntryPointNode = factory.MethodEntrypoint(method, getUnboxingStub);
+
+                Vertex methodSignature = GetNativeLayoutSignatureForMethodEntry(factory, method);
```

Comment:
>GetNativeLayoutSignatureForMethodEntry(factory, method); [](start = 41, length = 56)

This should be a callout to a vertex node implementation that is what was depended on.